server/web/routes/proposals.js

```diff
@@ -254,6 +254,7 @@ const register = function (server, options) {
           proposal.hasFeedback = feedbacks.length === proposal.reviewerIds.length;
           proposal.hasFeedback &= Boolean(proposal.reviewerIds.length);
           proposal.isAssignedToChair = proposal.reviewerIds.includes(user._id.toString());
+          proposal.history = await findAllProposalsInSameGroup(proposal);
```

Comment:
don't attach history object to proposal. 
Please first see my comments for findAllProposalInSmaeGroup. And attach the revisedProposals list to proposal.